src/main/java/pl/sda/library/infrastructure/upolujebooka/UpolujebookaScrapper.java

```diff
@@ -5,20 +5,62 @@
 import org.jsoup.nodes.Element;
 import org.jsoup.select.Elements;
 import org.springframework.stereotype.Component;
+import pl.sda.library.domain.model.Book;
 
+import java.io.BufferedReader;
 import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
 import java.util.ArrayList;
-import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Component
-class UpolujebookaScrapper {
-    private final static String URL = "https://upolujebooka.pl";
+public class UpolujebookaScrapper {
+    private final static String pageURL = "https://upolujebooka.pl";
+
+    public static Book findBookFromUrl(String url){
+        Book book = new Book();
+        try{
+            if (!url.isEmpty()) {
+                StringBuilder pageContent = new ReadPageContent(url).invoke();
+                Document documentURL = Jsoup.parse(pageContent.toString());
+                book.setAuthors((documentURL.select("div.authors > h2 >a").textNodes().stream()
+                        .map(Object::toString)
+                        .collect(Collectors.joining(","))));
+                book.setCategories(documentURL.select("div.container > ol > li > span > a").eachText().stream()
+                        .map(Object::toString)
+                        .collect(Collectors.joining(",")));
+                book.setTitle(documentURL.select("div.col-lg-6.col-md-8.col-sm-8 > h1").first().childNode(0).toString().trim());
+                book.setDescription(documentURL.select("div.description > p").text());
+                book.setCover(pageURL + "/" + documentURL.select(" Div.DetailImage > img ").first().attr("src"));
+            }
+        }catch (Exception exception) {
```

Comment:
można by KIEDYŚ pomyśleć o bardziej konkretnych wyjątkach albo swoich
bo exception to łapie wszystko